### DISCUSSION TITLE
Functions inside template directives?

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3515] [1. Help] [answered]
Within `Alpine.data('$', function(){ return {` I use `$search: { ['@input'](){ ...}, ['@keyup.enter'](){ ... }`. Since on both events do the same, and it is not possible to co-registger multiple events like `['@keyup.enter,@input']`, I wanted to declare a search function inside `$search`, but somehow this results in an error telling me that this is illegal (code).

How can I reuse the code for multiple search triggers?

**Comment by simonferndriger** [comment]
Figured out that the `@input` event works differently than I imagined so I just use a single event again. And when I place the function on the same level as `$search` it worked as well.

**Comment by ekwoka** [comment, marked as answer]
You would do it by declaring the function outside and binding it.
